clang/lib/Sema/SemaChecking.cpp

```diff
@@ -7974,8 +7977,18 @@ void CheckFormatHandler::CheckVarargsInCheckedScope(
           getSpecifierRange(StartSpecifier, SpecifierLen));
       }
 
-      // In printf-like functions, only allow scalar type arguments with format
-      // specifiers other than %s.
+    // In printf-like functions, only allow scalar type arguments with format
+    // specifiers other than %s.
+
+    // TODO: Currently, we do not handle the case where an out-of-bounds
+    // null-terminated array is passed as an argument to %s. The following code
+    // is allowed even though it might be a safety hole.
+    // _Checked {
+    //   _Nt_array_ptr<char> p : count(5);
+    //   printf("%s", p + 1234);
+    // }
+    // Issue https://github.com/microsoft/checkedc-clang/pull/1182 tracks this.
```

Comment:
I think you meant:
```suggestion
    // Issue https://github.com/microsoft/checkedc-clang/issues/1178 tracks this.
```
It's unfortunate that there doesn't seem to be an easier way to reuse the normal code path for validating an argument to a `const char * : itype(_Nt_array_ptr<const char>)` parameter, which should do this for free.